hazelcast/src/main/java/com/hazelcast/map/impl/SizeEstimator.java

```diff
@@ -16,18 +16,101 @@
 
 package com.hazelcast.map.impl;
 
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.nio.UnsafeHelper;
+import sun.misc.Unsafe;
+
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import javax.management.openmbean.CompositeDataSupport;
+import java.lang.management.ManagementFactory;
+
 /**
  * Size estimator general contract.
  *
  * @param <T> the type of object which's size going to be estimated.
  */
-public interface SizeEstimator<T> {
+public abstract class SizeEstimator<T> {
+
+    public static final boolean HAS_COMPRESSED_OOPS = hasCompressedOops();
+
+    @SuppressWarnings("CheckStyle")
+    public static final int OBJECT_HEADER_IN_BYTES = HAS_COMPRESSED_OOPS ? 12 : 16;
```

Comment:
not from spec. in fact more reliable way might be getting field offset of a sample field on a sample class. Such as

``` java
class Foo {
    int i;
}
```

in here, field offset of `i` should be the header size of object (I assume jvm impl doesn't have stupid layout mechanism).

Similar approach is generally used for getting shallow size of objects by getting max field offset and aligning it to heap word size (which is `8` by default)